src/librustdoc/clean/types.rs

```diff
@@ -338,14 +333,14 @@ crate struct Item {
     /// Information about this item that is specific to what kind of item it is.
     /// E.g., struct vs enum vs function.
     crate kind: Box<ItemKind>,
-    crate def_id: FakeDefId,
+    crate def_id: ItemId,
 
     crate cfg: Option<Arc<Cfg>>,
 }
 
 // `Item` is used a lot. Make sure it doesn't unintentionally get bigger.
 #[cfg(all(target_arch = "x86_64", target_pointer_width = "64"))]
-rustc_data_structures::static_assert_size!(Item, 48);
+rustc_data_structures::static_assert_size!(Item, 56);
```

Comment:
Can you box Auto and Blanket ids to avoid this size increase?

Reply:
That leads to a pretty huge change because I need to box the `ItemId` itself and now there are `*` everywhere, but the size is now 48 :sweat_smile:

Reply:
No, you don't need to box the whole ID - in fact you shouldn't, because it's used a lot. I was suggesting:
```rust
crate enum ItemId {
    DefId(DefId),
    Auto(Box<ImplId>),
    Blanket(Box<ImplId>),
    Primitive(CrateNum),
}
struct ImplId { trait_: DefId, for_: DefId }
```

Reply:
Oh I see! Thats much better

Reply:
That's still somewhat annoying, because now `ItemId` can't implement `Copy` and that is used in a *bunch* of places

Reply:
Can you paste the errors?

Reply:
I already fixed them. They were mostly about use of moved object.

Reply:
Actually I don't think we should box here. The assertion exists to prevent memory bloat; 48 and 56-sized objects will typically take part in the same arena region (i.e. both are stored in 60-64 byte blocks) so there won't be additional bloat. On the other hand, boxing guarantees that there will be additional allocation

Reply:
I added a commit that reverts the "Box up ItemId to decrease size", so you can test yourself, and it's still easy to switch between the solutions